_posts/2020-12-24-Bulletproofs+-in-Monero.md

```diff
@@ -0,0 +1,81 @@
+This is the preprint used as the basis for the proposed Monero protocol implementation.
+- [Bulletproofs+ code](https://github.com/SarangNoether/monero/tree/bp-plus) by Sarang Noether.
+This is the new implementation code written for compatibility with the Monero codebase.
+- [Consensus-related code](https://github.com/moneromooo-monero/bitmonero/tree/bp+c) by `moneromooo`. This code is necessary for a network upgrade that would include Bulletproofs+ proofs as a consensus rule.
+
+
+## Range proving in zero knowledge
+
+The Monero confidential transaction protocol requires the use of a zero-knowledge range proving system. Because inputs and outputs in Monero transactions have their value hidden, it's necessary to secretly prove that they represent valid amounts to avoid overflows that would fool the protocol's balance checks. The constructions used for range proving have evolved over time. Originally, the Monero protocol used a variation of ring signatures for this purpose; however, the resulting proofs were very large and slow to generate and verify, leading to slow synchronization of the blockchain and a large amount of chain bloat.
+
+This was overhauled after the release of Bulletproofs, a much more efficient range proving system. With Bulletproofs, range proofs are much smaller and faster to verify; further, multiple proofs can be verified at the same time in a batch, leading to even more efficient synchronization.
+
+A newer preprint modifies the Bulletproofs construction to produce Bulletproofs+, an even more efficient range proving system. Range proofs in Bulletproofs+ retain a similar underlying structure to those in Bulletproofs; however, they are slightly smaller, faster to generate, and faster to verify.
+
+Implementation code is now available that is compatible with the Monero codebase for easy deployment.
+
+
+## Efficiency
+
+Side-by-side efficiency comparisons between Bulletproofs and Bulletproofs+ range proofs are possible using the performance test framework in the Monero codebase.
+
+The size and timing characteristics of range proofs depend on the structure of the transaction that uses them. Because of the way that both the Bulletproofs and Bulletproofs+ algorithms work, the number of outputs in a transaction is effectively rounded up to the next power of two for range proving purposes, with a maximum of 16 outputs permitted in a transaction. The vast majority of Monero transactions contain two outputs, but 16 outputs is also common for pool payouts and other purposes.
+
+
+### Size
+
+"Regardless of the number of outputs in a transaction, the corresponding Bulletproofs+ range proof is *96 bytes* smaller than a Bulletproofs range proof."
+
+This table shows the reduction in size for the most common transaction types seen on the Monero network.
+
+Spent inputs | New outputs | Current size | New size | Reduction, % smaller
+--- | --- | --- | --- | ---
+1 | 2 | 1.42 kB | 1.33 kB | 6.6%
+2 | 2 | 1.92 kB | 1.83 kB | 5.1%
```

Comment:
I would suggest to use these tables less when possible. It's not a very mobile-friendly element.